Feature/dashboard event tracking

pattisdr opened the pull request:
:warning: Hold off review until dashboard settles down.
# Purpose

Issue https://openscience.atlassian.net/browse/OSF-5721
Adds high level Google Analytics event tracking to new dashboard to gauge what features are being used.
# Changes

<b>Tracks a combined 96 events in project organizer/dashboard and home page.</b>  All high level tracking, basically clicks, if users filtered, etc.  No information being tracked about specific users.

Event tracking format looks like this: 
ga('send', 'event', [eventCategory], [eventAction], [eventLabel], [eventValue], [fieldsObject]);

Using eventCategory, eventAction, eventLabel.  Can explore sending in eventValue with appropriate fields (must be an integer).

Sample event: 
   `$osf.trackClick('recentActivity', 'filter', 'filter-on-files');`

This indicates that someone within the recent activity widget clicked on the word "files" to filter log actions by files.
# For the reviewer

Clicking on some links trigger the same actions and I've chosen to group them together.  For example, there are two places where you can click to add a component on the dashboard. I've grouped them together. Going through the events and determining if there is a better way to group them would be important.

Also, the vast majority of the events that can be tracked are here, but I am sure I am missing some.  Are there any not included that we need?

Ghalko committed:
Merged branch feature/admin_node into feature/admin_spam_user

Ghalko committed:
Add spam list for user

- Connect user info page to spam lists

Ghalko committed:
Disable user

Except for mailchimp of course.

[skip ci]

Ghalko committed:
Refactor user search/user display

Because it wasn't making sense downstream.

Ghalko committed:
Use table for users

Ghalko committed:
Refactor Node

Separate out node search from node details display

Ghalko committed:
Fix tests

Ghalko committed:
Merge branch 'develop'

Conflicts:
	admin/spam/serializers.py
	admin/templates/spam/comment.html

Ghalko committed:
Merge branch 'feature/admin_spam_user'

Conflicts:
	admin/templates/users/user.html
	admin/users/urls.py

Ghalko committed:
Add modal

Ghalko committed:
Disable user

user.save() really?

Ghalko committed:
Reactivate user

Ghalko committed:
Remove two factor with click

samanehsan committed:
Pass file guids to waterbutler

In the waterbutler callback for OSF
authentication, if the action is "movefrom",
pass the guid of the file being affected to
waterbutler. If a folder is being moved or
renamed, pass the guids of all of it's files
(inluding files in subfolders).

Ghalko committed:
Add confirm modal

Ghalko committed:
Refactor links

Use reverse

samanehsan committed:
Update Guid.referent on file move/rename

Currently, whenever the path or node of a file changes,
a new StoredFileNode will be created and the file will
receive a new Guid.

Update the file's  Guid.referent to the new StoredFileNode
so that the Guid of the file remains the same. If the node
of the file changes, update the `node` field for all of it's
comments to the destination_node.

samanehsan committed:
Add tests for file comments move/rename

samanehsan committed:
Handle nested folders moving providers

samanehsan committed:
Add tests for cross-provider folder moves

samanehsan committed:
Add nose_parameterized to requirements.txt

samanehsan committed:
Fix failing tests

Make file comment mixin inherit from
object instead of OsfTestCase because
those tests will get run and fail otherwise.

pattisdr committed:
Merge branch 'feature/logwidg3' of https://github.com/hmoco/osf.io into combined_dash

Conflicts:
	website/templates/home.mako

samanehsan committed:
Skip tests for folder move/rename

Since the waterbutler payloads for
when addon folders are moved or
renamed are inconsistent -- some actions
include a "children" key with the folder
contents for some providers while others
do not -- skip these tests for now.

samanehsan committed:
Return empty list instead of None if no guids

samanehsan committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/pass-guids-to-wb

Conflicts:
	website/project/views/comment.py

samanehsan committed:
Ensure deleted file guid not passed to wb

In FileNode.get_file_guids, when checking if
the moved/renamed folder contains any files
with guids, also check that those files
are not deleted before sending their guids
to waterbutler.

pattisdr committed:
index on combined_dash: 553ba04 Merge branch 'feature/logwidg3' of https://github.com/hmoco/osf.io into combined_dash

pattisdr committed:
WIP on combined_dash: 553ba04 Merge branch 'feature/logwidg3' of https://github.com/hmoco/osf.io into combined_dash

Ghalko committed:
Merged branch develop into feature/admin_spam_user

Ghalko committed:
Respond to CR

Ghalko committed:
Merged branch develop into feature/admin_deactivate_2factor

Ghalko committed:
Respond to CR

Ghalko committed:
Add tests

samanehsan committed:
Get guid_list post file/folder move/rename

When a box folder is moved/renamed, it is
not possible to find its contents with just
the folder path. But, since the materialized
path is not available until after wb gets auth
from the OSF, get the list of file guids in
update_file_guid_referent & update the get_guids
method to take in the materialized path.

samanehsan committed:
Move deleted addon files to TrashedFileNode

When an addon file is deleted, make a TrashedFileNode
record and delete the StoredFileNode record. If an
addon folder is deleted, do this for all of its files.

If a filenode that does not exist is being viewed,
move it to TrashedFileNode & remove its StoredFileNode
as well -- this handles files deleted through their
external provider.

Ghalko committed:
Merge branch 'develop'

Conflicts:
	admin/base/forms.py
	admin/base/views.py
	admin/nodes/serializers.py
	admin/nodes/templatetags/node_extras.py
	admin/nodes/urls.py
	admin/nodes/views.py
	admin/templates/base.html
	admin/templates/nodes/node.html
	admin/templates/users/user.html
	admin/users/serializers.py
	admin/users/templatetags/user_extras.py
	admin/users/urls.py
	admin/users/views.py
	admin_tests/nodes/test_serializers.py
	admin_tests/nodes/test_views.py
	admin_tests/users/test_serializers.py
	admin_tests/users/test_views.py
	admin_tests/utilities.py

Ghalko committed:
Add tests

pattisdr committed:
Add functions to sent in a hash of the user_id and the timestamp alongside google analytics information.

pattisdr committed:
Add event tracking to QuickSearch projects.

pattisdr committed:
Add trackClick method for google analytics event tracking in the new dashboard.

pattisdr committed:
Add event tracking to logTextParser for when user navigates to a user or a project from activity logs.

pattisdr committed:
Add event tracking to Recent Activity Widget.

pattisdr committed:
Add event tracking to New and Noteworthy module.

pattisdr committed:
Add event tracking to meetings and conferences.

pattisdr committed:
Add event tracking to add project plugin.

pattisdr committed:
Add event tracking to project organizer.

pattisdr committed:
Add event tracking to dashboard.js.

pattisdr committed:
Merge cleanup.

samanehsan committed:
Test addon_delete_file_node

Test that the StoredFileNode of a deleted
file is removed, and that a TrashedFileNode
is created, when an addon file is deleted.

samanehsan committed:
Exclude osfstorage files from addon_delete_file_node

Since osfstorage files already get trashed on delete,
do not delete them in addon_delete_file_node

samanehsan committed:
Fix flake8 error

Ghalko committed:
Merge branch 'develop'

Conflicts:
	admin/spam/urls.py
	admin/spam/views.py
	admin/templates/spam/comment.html
	admin/templates/spam/spam.html

samanehsan committed:
Fix failing tests & test get_file_guids

pattisdr committed:
Add event tracking for clicking on breadcrumbs.

pattisdr committed:
Add event tracking for removing project from collection.

pattisdr committed:
Add event tracking for reloading projects.

pattisdr committed:
Add event tracking for items on mobile view.

samanehsan committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/pass-guids-to-wb

samanehsan committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/pass-guids-to-wb

Conflicts:
	tests/test_comments.py

pattisdr committed:
Merge dashboard-redesign-2.

pattisdr committed:
Merge branch 'dashboard-redesign-2' of https://github.com/caneruguz/osf.io into feature/dashboard_event_tracking

Conflicts:
	website/static/js/meetingsAndConferencesPlugin.js
	website/static/js/newAndNoteworthyPlugin.js
	website/static/js/quickProjectSearchPlugin.js

billyhunt committed:
add sign up link to sine in page

billyhunt committed:
remove newline

pattisdr committed:
Add ability to to add-project event tracking to determine if coming from quickSearch or from myProjects.

pattisdr committed:
Merge branch 'dashboard-redesign-2' of https://github.com/caneruguz/osf.io into feature/dashboard_event_tracking

pattisdr committed:
Change 'dashboard' category to 'myProjects' for event tracking.

pattisdr committed:
Merge branch 'feature/500_error_when_accessing_log_detail' into feature/dashboard_event_tracking

pattisdr committed:
Restore m.redraw().

pattisdr committed:
Merge branch 'dashboard-redesign-2' of https://github.com/caneruguz/osf.io into feature/dashboard_event_tracking

Conflicts:
	website/static/js/quickProjectSearchPlugin.js

billyhunt committed:
change link to a button to sign up

billyhunt committed:
make sign up window default with click of signup button via campaign

billyhunt committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/add-sign-up-link-to-sign-in

* 'develop' of https://github.com/CenterForOpenScience/osf.io:
  Fix improper usages of CommentFactory
  Only update last_login when not at an API url
  Hide terms when submitting form
  Remove unnecessary ternary; use constant
  Add test for more when there are 2 short wiki pages
  Update more variable to incl other wiki page
  Change min length count in view to match md truncate
  Add read more link in new div

Ghalko committed:
Respond to CR

Ghalko committed:
Merge branch 'develop'

Conflicts:
	admin/base/views.py
	admin/nodes/serializers.py
	admin/nodes/templatetags/node_extras.py
	admin/nodes/urls.py
	admin/nodes/views.py
	admin/spam/urls.py
	admin/spam/views.py
	admin/templates/base.html
	admin/templates/nodes/node.html
	admin/templates/users/user.html
	admin/users/serializers.py
	admin/users/templatetags/user_extras.py
	admin/users/urls.py
	admin/users/views.py
	admin_tests/nodes/test_serializers.py
	admin_tests/nodes/test_views.py
	admin_tests/users/test_serializers.py
	admin_tests/users/test_views.py
	admin_tests/utilities.py

Commit:
Testing if SHA5 description works on revisions

Commit:
revisions testing

samanehsan committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/pass-guids-to-wb

samanehsan committed:
Fix file serializer comments link

The file serializer comments relationship
field should use the file guid in the url
instead of the file id.

samanehsan committed:
Make non-osfstorage addons commentable

samanehsan committed:
Update comments on file node change

When a file is moved to a new node, update
it's comments with the new node. Also
uncomment tests for checking that the comment
remains on files post folder/file move/rename.

Commit:
testing revisions again

Commit:
revert changeds

samanehsan committed:
Make get_file_guids handle TrashedFileNodes

When an osfstorage file is moved to a different
provider, the StoredFileNode gets deleted and
a TrashedFileNode is created. In order to get the
guid of the osfstorage file post-move, look up
the TrashedFileNode, then search for the guid that
has that referent.

samanehsan committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/pass-guids-to-wb

Ghalko committed:
Merge branch 'develop'

Conflicts:
	admin/base/views.py
	admin/nodes/serializers.py
	admin/nodes/templatetags/node_extras.py
	admin/nodes/urls.py
	admin/nodes/views.py
	admin/spam/urls.py
	admin/spam/views.py
	admin/templates/base.html
	admin/templates/nodes/node.html
	admin/templates/users/user.html
	admin/users/serializers.py
	admin/users/templatetags/user_extras.py
	admin/users/urls.py
	admin/users/views.py
	admin_tests/nodes/test_serializers.py
	admin_tests/nodes/test_views.py
	admin_tests/users/test_serializers.py
	admin_tests/users/test_views.py
	admin_tests/utilities.py

Commit:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/mdhashtips

samanehsan committed:
Fix file serializer comments link

Since the file comments relationship
field is a link to the node comments
endpoint filtered by the file guid,
if the file does not have a guid (or any
other filters for that link) make the

samanehsan committed:
Remove file_guid from node.commented_files on delete

Commit:
Create popover for MD5 and SHA2

Ghalko committed:
Catch runtime error

Ghalko committed:
Write tests

samanehsan commented on website/static/js/pages/home-page.js:
```diff
@@ -6,6 +6,7 @@
 
 var $ = require('jquery');
 var m = require('mithril');
+var $osf = require('js/osfHelpers');
```
This file already requires `js/osfHelpers` a few lines up, so this can be removed.

samanehsan commented on api/logs/permissions.py:
```diff
@@ -12,8 +12,14 @@ class ContributorOrPublicForLogs(permissions.BasePermission):
-        for node_id in obj._backrefs['logged']['node']['logs']:
-            node = get_object_or_error(Node, node_id, display_name='node')
-            if ContributorOrPublic().has_object_permission(request, view, node):
+        if obj._backrefs.get('logged'):
```
Was this backrefs change meant to be a part of a different PR?

samanehsan replied:
Oh, it looks like it's from a commit that was merged into `caneruguz/dashboard-redesign-2` Can you please bring this up to date with that branch @pattisdr?

pattisdr replied:
Unless @caneruguz has told you differently, I was told to hold off on event tracking for a few days(as the dashboard itself is changing very rapidly right now).  Caner, @hmoco, and I are responding to Steve's review right now.  When the dashboard is more settled, then I will bring this up to date.  I think the review should wait until then.

pattisdr replied:
For example, it looks like the recent activity widget was removed from the home page for now, so all of my event tracking of that module needs to go.  But I will wait until I see what else is changing.

samanehsan replied:
Oh ok, that makes sense! Thanks for letting me know!

pattisdr replied:
Yeah, I should have put a note on this!

tonynghieu committed:
Revise the issues from initial pull request

billyhunt committed:
Change button color to green

billyhunt committed:
reorder buttons, change color, fix login size height issue

billyhunt committed:
remove unnecessary css file and some spacing

Ghalko committed:
Default fixed

Fixed associated tests

samanehsan committed:
Add docstring and fix formatting

samanehsan committed:
Remove subfolders when addon folder is deleted

When an addon folder is deleted, in addition to
moving the files to TrashedFileNode, delete
any subfolders from StoredFileNode.

samanehsan committed:
Minor rename/refactor

billyhunt committed:
remove campaign and pass sign up variable in url

caneruguz committed:
Fix All my project and registrations should not accept drop

caneruguz committed:
If returned nodes are errors omit from flattening

billyhunt committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into feature/add-sign-up-link-to-sign-in

* 'develop' of https://github.com/CenterForOpenScience/osf.io: (42 commits)
  Minor logic simplification
  take out multidashes from BAPI docs to appease swagger
  change contributions to contribution when there is only one
  Changing a test case to check for the correct string
  remove rescale_ratio from code
  Changing a test to check for a the correct string
  code cleanup
  furthur fix the logic for the warning and welcome banner
  [OFI-25] Updated department heading in employment tab to read "Department / Institute"
  Remove "Project: " and "Component: " from node titles in project manager
  Minor change to variable name and language
  add note to manually create index on user emails
  Revert "Add indices to the user collection"
  Add indices to the user collection
  fix the error and fix flake8
  [#OSF-5827]. remove the welcome message and add the red warning
  fix incorrect file update api docs
  Add note about hiring
  close old meetings
  Add readme
  ...

billyhunt committed:
autofocus form based on if you are logging in or signing up

pattisdr committed:
Merge branch 'dashboard-redesign-2' of https://github.com/caneruguz/osf.io into feature/dashboard_event_tracking

Conflicts:
	website/static/js/home-page/newAndNoteworthyPlugin.js
	website/static/js/home-page/quickProjectSearchPlugin.js
	website/static/js/pages/home-page.js
	website/static/js/recentActivityWidget.js

samanehsan committed:
Fix waterbutler 500 error when getting dropbox file

Check if the referent.provider is dropbox and
change the referent from a StoredFileNode to a
DropboxFile before making a request to waterbutler.

icereval committed:
add rollback error handling to transaction middleware

samanehsan committed:
Add docstring, TODO and minor refactors

Minor refactors, add TODO for removing
skipped tests and docstring for creating file
helper.

Also add list of the id-based addons that track
where a file is (osfstorage and box) to
settings instead of hard-coding providers.

icereval committed:
flake8

chennan47 committed:
Fix the error of contributors display for a public project under anonymous link

sloria committed:
Merge branch 'hotfix/0.65.6' into develop

billyhunt committed:
change 1 to true

sloria committed:
Change tooltip text

according to https://github.com/CenterForOpenScience/osf.io/pull/5148#issuecomment-194496121

sloria committed:
Merge branch 'tnghieu-feature/mdhashtips' into develop

billyhunt committed:
remove conditional and set value as false if not true

sloria committed:
Merge branch 'hotfix/0.65.7' into develop

caneruguz committed:
Empty panel if no item is selected and reset selected items if no nodes are displayed

caneruguz committed:
Improve paddings [skip ci]

sf2ne committed:
Improvement onto icon images: Added ARIA labels to social media icons in footer to make them readable by voiceover feature in safari

Commit:
Changed the format of the text generated by hovering over permission in project-> contributors to look more professional.

sloria committed:
Refactor autofocus and remove unused variable

sloria committed:
Merge branch 'billyhunt-feature/add-sign-up-link-to-sign-in' into develop

caneruguz committed:
Fix minor border in project organizer

caneruguz committed:
Decrement count on collections at node removal

caneruguz committed:
Run make Tree function only when the all my projects is loading
otherwise it's breaking counts and displays

caneruguz committed:
Restore user node count
because the maketree problem was what was causing this issue

samanehsan committed:
Split up update_file_guid_referent

Move logic for creating a new file into
a helper function.

caneruguz committed:
Fix project vs component language continuity in the add project plugin

sloria committed:
Merge pull request #5104 from samanehsan/feature/addon-trashed-file-nodes

Feature: Create trashed addon file nodes

sloria committed:
Merge pull request #5166 from sf2ne/iconLabels

Improvement onto icon images: Added ARIA labels to social media icons

caneruguz committed:
Sort projects by last modified as default

sloria committed:
Merge pull request #5040 from Ghalko/feature/admin_user_disable

[Feature] Admin Disable User

caneruguz committed:
Fix bug with all items receiving format data for PO

caneruguz committed:
Fix collection content count increasing even when addition fails

caneruguz committed:
Fix  collection pagination not flipping when all items in one page are removed.

caneruguz committed:
Go to last page after adding a collection

sloria committed:
Merge branch 'Ghalko-feature/admin_spam_user' into develop

sloria committed:
Merge branch 'develop' of https://github.com/CenterForOpenScience/osf.io into develop

caneruguz committed:
Make projects default view in narrow view

sloria committed:
Refactor to use has_addon

sloria committed:
Merge branch 'Ghalko-feature/admin_deactivate_2factor' into develop

sloria committed:
Merge pull request #5168 from caneruguz/feature/dashboard-fixes

Feature/dashboard fixes 2

sloria committed:
Merge pull request #5059 from samanehsan/feature/pass-guids-to-wb

Get & Update guid referents & add addon file comments

sloria committed:
Merge branch 'caneruguz-dashboard-redesign-2' into feature/dashboard

sloria committed:
Merge branch 'release/0.66.0' into feature/dashboard

sloria committed:
Fix outdated tests

pattisdr committed:
Merge branch 'feature/dashboard' of https://github.com/CenterForOpenScience/osf.io into feature/dashboard_event_tracking

Conflicts:
	website/static/js/addProjectPlugin.js
	website/static/js/home-page/newAndNoteworthyPlugin.js
	website/static/js/home-page/quickProjectSearchPlugin.js
	website/static/js/myProjects.js

pattisdr committed:
Modify event-tracking to accommodate recent dashboard changes.  Splitting out add-project into add-project and add-component. Changing names of event labels to match new button text, etc.

pattisdr commented:
Going to make this a PR to feature/dashboard instead now that dashboard-redesign-2 is merged.
